pkg/release/repository.go

```diff
@@ -108,6 +108,10 @@ func (r *Repo) CheckState(expOrg, expRepo, expBranch string) error {
 	if branch != expBranch {
 		return errors.Errorf("branch %q expected but got %q", expBranch, branch)
 	}
+	if branch != "kubernetes/release:master" && nomock == true {
```

Comment:
I think we have to include checks for `expOrg, expRepo, expBranch`, because `branch` will only contain the repository branch name.